zero_bin/common/src/pre_checks.rs

```diff
@@ -0,0 +1,23 @@
+use proof_gen::proof_types::GeneratedBlockProof;
+
+pub fn check_previous_proof_and_checkpoint(
+    checkpoint_block_number: u64,
+    previous_proof: &Option<GeneratedBlockProof>,
+    start: u64,
+) -> anyhow::Result<()> {
+    if let Some(proof) = previous_proof {
+        if proof.b_height + 1 != start {
+            return Err(anyhow::Error::msg(format!(
+                "Found previous b_height {} whereas range start is {}",
+                proof.b_height, start,
+            )));
+        }
+    } else if checkpoint_block_number >= start {
```

Comment:
this shouldn't be an else if

Reply:
also I you tagged my previous comment as resolved but it is not, we shouldn't just check whether `checkpoint_block_number >= start` (the comparison is incorrect btw, checkpoint_block_number == start _is_ invalid)

Reply:
i am not sure i understand correctly
checkpoint_block_number can be equal to start_block?

Reply:
Nevermind, I had misread the code. Yes it is fine, just the `else if` should be removed.